examples/nlp/gpt/conf/gpt_self_rewarding.yaml

```diff
@@ -0,0 +1,281 @@
+defaults:
+  - optional tp_overlap@model.ub_tp_comm_overlap_cfg:
+
+trainer:
+  num_nodes: 8
+  devices: 8
+  accelerator: gpu
+  precision: bf16
+  
+  # self-rewarding specific args
+  self_rewarding:
+    max_iterations: 3   # the number of iterations to conduct self-rewarding for, with the reference policy updated on each iteration
+    max_epochs: 1       # the number of epochs to use per iteration
+    max_steps: -1
+    val_check_interval: 0.1
+    save_interval: 100
+    limit_train_batches: 1.0
+
+    # how many GBS we loop over
+    limit_val_batches: 1.0
+    gradient_clip_val: 1.0
+
+    # Accelerate training times by accelerating inference stage using TRTLLM
+    trt_llm:
+      enable: False
+      # reshard is not supported in Self-Rewarding
+
+      # TRTLLM preallocates activation memory according to the number of input tokens
+      # By default, assume the max input length is half of the model sequence length
+      max_input_len: ${int_div:${model.encoder_seq_length}, 2}
+      generation_batch_size: ${model.spin.rollout_micro_batch_size}
+
+      model_type: gptnext
+
+      # Unload and reload the TRTLLM engine before and after the training stage 
+      # Reloading the engine incurs a constant time overhead
+      unload_engine_train: False
+
+  # do not change these
+  logger: False # logger provided by exp_manager
+  enable_checkpointing: False
+  use_distributed_sampler: False
+  max_time: null
+  max_epochs: ${.self_rewarding.max_epochs}
+  max_steps: ${.self_rewarding.max_steps}
+
+exp_manager:
+  explicit_log_dir: /results
+  exp_dir: null
+  name: megatron_gpt
+  max_time_per_run: ${trainer.max_time}
+  create_wandb_logger: False
+  wandb_logger_kwargs:
+    project: nemo_aligner_dpo
+    name: rlhf_gpt3_dpo
+  resume_from_checkpoint: null # The path to a checkpoint file to continue the training, restores the whole state including the epoch, step, LR schedulers, apex, etc.
+  resume_if_exists: True
+  resume_ignore_no_checkpoint: True
+  create_checkpoint_callback: True
+  checkpoint_callback_params:
+    monitor: val_loss
+    save_top_k: 3
+    mode: min
+    always_save_nemo: False # saves nemo file during validation, not implemented for model parallel
+    save_nemo_on_train_end: True # not recommended when training large models on clusters with short time limits
+    filename: 'megatron_gpt--{${.monitor}:.3f}-{step}-{consumed_samples}-{iteration}-{epoch}'
+    model_parallel_size: ${multiply:${model.tensor_model_parallel_size}, ${model.pipeline_model_parallel_size}}
+
+pretrained_checkpoint:
+  restore_from_path: null
+
+model:  
+  mcore_gpt: True
+  micro_batch_size: 1
+  global_batch_size: 64
+  megatron_amp_O2: True
+  dist_ckpt_format: zarr
+
+  spin:
+    log_prob_forward_micro_batch_size: ${model.micro_batch_size}
```

Comment:
Won't that OOM?

Reply:
That depends -- it can OOM indeed which is why it's an option that someone can tweak independently if needed. I don't feel super strongly about it, but it seems more natural to me to use the same defaults as PPO for similar options.

Reply:
We can do it, but I think it would need to be rollout_mbs // 2, because the underlying model uses the new rewrite where it gets * 2 to account for the chosen + rejected

Reply:
I have implemented it with divide 2